### PR TITLE
Refactor area, line and point in mark.

### PR DESCRIPTION
**Pull request by bobocandys**
Addressing #1284 except for `bar`

- [x] area
- [x] line
- [x] point
- [x] text
- [x] tick

### CODE REVIEW

**Comment by kanitw** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/vega/vega-lite/pull/1403'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by bobocandys** [comment]
@kanitw

File: src/compile/mark/area.ts [hunk 1]
```diff
@@ -11,80 +14,107 @@ export namespace area {
...
 
-    const orient = model.config().mark.orient;
-    if (orient !== undefined) {
-      p.orient = { value: orient };
-    }
+    const _orient = orient(config.mark.orient);
+    if (_orient) { p.orient = _orient; }
+    
+    const _x = x(model.encoding().x, model.scaleName(X), model.stack());
+    if (_x) { p.x = _x; }
```
Comment (kanitw) [inline src/compile/mark/area.ts]: I think `x` is always defined (Otherwise we can't draw an area)

File: src/compile/mark/area.ts [hunk 2]
```diff
@@ -11,80 +14,107 @@ export namespace area {
...
+
+    const _x2 = x2(model.encoding().x, model.scaleName(X), model.stack(), config.mark.orient);
+    if (_x2) { p.x2 = _x2; }
+    
+    const _y = y(model.encoding().y, model.scaleName(Y), model.stack());
+    if (_y) { p.y = _y; }
```
Comment (kanitw) [inline src/compile/mark/area.ts]: I think `y` is always defined (Otherwise we can't draw an area)

File: src/compile/mark/area.ts [hunk 3]
```diff
@@ -11,80 +14,107 @@ export namespace area {
...
+  }
+  
+  function orient(orient: string): VgValueRef {
+    if (orient) {
+      return { value: orient };
+    }
+    return undefined;
+  }
+  
+  function x(fieldDef: FieldDef, scaleName: string, stack: StackProperties): VgValueRef {
```
Comment (kanitw) [inline src/compile/mark/area.ts]: I think we need to pass in scale and if (!scale), then don't output `scaleName`.

File: src/compile/mark/area.ts [hunk 4]
```diff
@@ -11,80 +14,107 @@ export namespace area {
...
+    } else if (isDimension(fieldDef)) {
+      return {
+        scale: scaleName,
+        field: field(fieldDef, { binSuffix: '_mid' })
       };
     }
+    return undefined;
+  }
 
+  function x2(fieldDef: FieldDef, scaleName: string, stack: StackProperties, orient: string): VgValueRef {
```
Comment (kanitw) [inline src/compile/mark/area.ts]: I think we need to pass in scale and if (!scale), then don't output `scaleName`.

File: src/compile/mark/area.ts [hunk 5]
```diff
@@ -11,80 +14,107 @@ export namespace area {
...
+        return {
+          scale: scaleName,
           value: 0
         };
       }
     }
+    return undefined;
+  }
 
+  function y(fieldDef: FieldDef, scaleName: string, stack: StackProperties): VgValueRef {
```
Comment (kanitw) [inline src/compile/mark/area.ts]: I think we need to pass in scale and if (!scale), then don't output `scaleName`.

File: src/compile/mark/area.ts [hunk 6]
```diff
@@ -11,80 +14,107 @@ export namespace area {
...
+      return {
+        scale: scaleName,
+        field: field(fieldDef, { binSuffix: '_mid' })
       };
     }
-
+    return undefined;
+  }
+  
+  function y2(fieldDef: FieldDef, scaleName: string, stack: StackProperties, orient: string): VgValueRef {
```
Comment (kanitw) [inline src/compile/mark/area.ts]: I think we need to pass in scale and if (!scale), then don't output `scaleName`.

**Comment by kanitw** [comment]
test

**Comment by kanitw** [comment]
test

**Pull request merged**